app/controllers/listings_controller.rb

```diff
@@ -6,12 +8,8 @@ class ListingsController < ApplicationController
   def create
     @listing = current_user.listings.new(listing_params)
     @tmdb_id = params[:tmdb_id]
+    @movie = Movie.find_by(tmdb_id: @tmdb_id) || tmdb_handler_add_movie(@tmdb_id)
```

Comment:
This line prefers the database movie object over hitting the TMDB api. You'll see this pattern through all of these controller actions.